lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -150,7 +150,7 @@ LogicalResult CreateSiFiveMetadataPass::emitMemoryMetadata() {
             auto parentModule = inst->getParentOfType<FModuleOp>();
             if (dutMod == parentModule)
               hierName = parentModule.getName().str();
-            hierName = hierName + "." + inst.name().str();
+            hierName += (Twine(".") + inst.name()).str();
```

Comment:
would this avoid another string as:
```
hierName = (Twine(hierName) + "." + inst.name()).str();
```

Reply:
It may also be better to switch to a `SmallString hiername` here and just use `hiername.append(...)`.

Reply:
Okay, no need to create the intermediate string before appending, right? Updated.

Reply:
Oh, hmm didn't see the second comment.  Indeed.